rwiki/rwiki-impl/impl/src/java/uk/ac/cam/caret/sakai/rwiki/component/service/impl/RenderServiceImpl.java

```diff
@@ -137,11 +137,13 @@ public String renderPage(RWikiObject rwo, String pageSpace,
 				}
 				if (!showPage) {
 					content = content.substring(0,startRwikiName) + content.substring(finishRwikiName+1);
+					startRwikiName = content.indexOf("[", startRwikiName);
```

Comment:
Thank you very much @jesusmmp.

This always feel unreliable because it's missing comments. @JuanDavid102 can you review this indexes stuff?

I'd like some simple comments like....

// startRwikiName is the index of the page link bla bla Example: This is a [new page] startRwikiName = X

I mean, reviewing the PR I can't figure out if it's correct or not because I don't know the logic in depth, I can understand why, but not 100% sure!

Thanks

Reply:
Buenas @mpellicer te comento:

Cuando el código entra por la condición `if (!showPage) {` lo que hace `content = content.substring(0,startRwikiName) + content.substring(finishRwikiName+1);` es cortar content solamente mostrando así las paginas que puede leer. 
Por tanto tiene sentido que el nuevo comienzo donde empezar a buscar `startRwikiName` sea en la posición donde se encontraba el anterior. 
En cambio cuando si que puede leer el contenido, entrando por el `else` lo que hace es añadir a content "(*Esta página está disponible únicamente a los grupos:(2350) GRUPO 6, (2350) GRUPO 3)" y por tanto el nuevo comienzo debe ser el fin del anterior enlace `startRwikiName = content.indexOf("[", finishRwikiName);`

Gracias, un saludo.